src/FuncSharp.Examples/IOption.cs

```diff
@@ -0,0 +1,98 @@
+        }
+
+        private void HandlingNestedOptionsWithFlatMap(decimal number, decimal divisor)
+        {
+            IOption<decimal> divisionResult = Divide(number, divisor);
+            IOption<IOption<decimal>> resultOfDoubleDivision = divisionResult.Map(r => Divide(r, divisor));
+            IOption<decimal> flattenedResultOfDoubleDivision1 = resultOfDoubleDivision.Flatten(); // This option has value if both the inner and the outer option have value.
+            // Same can be done with 1 call.
+            IOption<decimal> flattenedResultOfDoubleDivision2 = divisionResult.FlatMap(r => Divide(r, divisor));
+        }
+
+        private void UsingOptionValueWithMatch(decimal number, decimal divisor)
+        {
+            var divisionResult = Divide(number, divisor);
+            // This overload takes 2 Func parameters. Each of those have to return a value and result is stored in the `roundedResult` variable.
+            decimal roundedResultOrFourteen = divisionResult.Match(
+                result => Math.Round(result),
+                _ => 14
+            );
+
+            // This overload accepts 2 optional void lambdas. If lambda isn't provided, nothing happens for that case.
+            divisionResult.Match(
+                result => Console.Write($"Division successful, result is: {result}."),
+                _ => Console.Write("Division failed, must have divided by zero.")
+            );
+            divisionResult.Match(result => Console.Write($"Division successful, result is: {result}."));
+            divisionResult.Match(ifSecond: _ => Console.Write("Division failed, must have divided by zero."));
+        }
+
+        private void GettingOptionValue(decimal number, decimal divisor)
+        {
+            var divisionResult = Divide(number, divisor);
+
+            // `Get` will throw an exception in case of empty option. Therefore you should not really use it at all.
+            // You should rather use Match to branch your code into individual cases where each case is guaranteed to work.
+            decimal valueOrExceptionThrown = divisionResult.Get();
+
+            decimal valueOrFallback1 = divisionResult.GetOrZero();
+            decimal valueOrFallback2 = divisionResult.GetOrElse(114m);
+            decimal valueOrFallback3 = divisionResult.GetOrElse(_ => 114m); // Lazy. Will only run the lambda if it needs to.
+
+            // These two are identical. Just Map creates one extra instance of Option for no reason.
+            decimal roundedDivisionResult1 = divisionResult.Map(r => Math.Round(r)).GetOrZero();
+            decimal roundedDivisionResult2 = divisionResult.Match(
+                r => Math.Round(r),
+                _ => 0
+            );
+        }
+
+        private void HandlingCollectionsOfOptions()
+        {
+            var numbers = new List<int> { 0, 1, 2, 3 };
+            var divisors = new List<int> { 0, 1, 2, 3 };
```

Comment:
You can turn this into parameter - list of tuples, so that it works in similar fashion to other methods